clang/lib/AST/Expr.cpp

```diff
@@ -3893,9 +3893,12 @@ namespace {
     }
 
     void VisitCXXBindTemporaryExpr(const CXXBindTemporaryExpr *E) {
-      if (E->getTemporary()->getDestructor()->isTrivial()) {
-        Inherited::VisitStmt(E);
-        return;
+      if (const CXXDestructorDecl *DtorDecl =
+              E->getTemporary()->getDestructor()) {
+        if (DtorDecl->isTrivial()) {
+          Inherited::VisitStmt(E);
+          return;
+        }
```

Comment:
As you already pointed out, this does seem like a weird fix for this. I’m not sure this is right. The documentation for `CXXBindTemporaryExpr` states:
```
/// This ensures the destructor is called for the temporary. It should only be
/// needed for non-POD, non-trivially destructable class types.
```
To me at least, this sounds like we should never get here with the destructor being null. Perhaps another approach would be to simply ignore the `=` if the next token is `{` and parse a function body. 

If the next token is not `{`, then we could ignore everything following it up to the next semicolon (unless it’s `= delete`, `= default`, or `= 0`, of course), i.e. essentially treat it as though it were `~Foo();`—if that’s not too complicated. And if we’re going with that, `assert`ing that the destructor is never null here is probably a good idea.

Reply:
I’ve been looking at some of the other uses of `CXXTemporary`, and it does seem like there are other places where we check for it being null, but it also doesn’t seem to be consistent (at least I’m not seeing a pattern here), and doing so also seems a bit contrary to the documentation here.

Reply:
The destructor seems to be invalid, we probably should add a validity check....somewhere

Reply:
It looks like it's normal for `nullptr` to slip into `CXXTemporary` as destructor:
https://github.com/llvm/llvm-project/blob/9145514fde484916971e6bb147c18f9235a9f2b5/clang/lib/Sema/SemaExprCXX.cpp#L7611-L7630
So the check I added might be a check in the right place.

Reply:
I think I'm happy with doing it it here, but it would be helpful to add a comment.
If someone has a better idea and want to inspects all places where CXXBindTemporaryExpr is created, we can do that in a subsequent PR.

Reply:
In addition to adding a comment here, thoughts about adding one to `CXXTemporary` to indicate that the destructor can be null? Because at least the comment above `CXXBindTemporaryExpr` makes it sound like that isn’t really supposed to be the case.

Reply:
Comments added.